src/port1.0/portlint.tcl

```diff
@@ -557,9 +684,18 @@ proc portlint::lint_main {args} {
     }
 
     if {[info exists checksums]} {
-        if {![regexp {size\s+(\d+)} $checksums]} {
-            ui_warn "Checksum(s) should include the 'size' field"
-            incr warnings
+        set checksum_lint [portlint::lint_checksum $checksums]
+
+        foreach err [lindex $checksum_lint 0] {
+            ui_error $err
+            incr errors
+        }
+
+        if {[llength [lindex $checksum_lint 0]] == 0} {
```

Comment:
Warnings should be shown regardless of whether there were errors.

Reply:
One of the reasons is because we stop processing immediately in an error situation, especially in the case of invalid tokens that don't make any sense for checksum.  In that situation, sometimes the warnings are inaccurate.

Reply:
OK, so don't show the ones that might be spurious then. Are there others that are still valid in that case though?

Reply:
OK, this has been resolved.